src/app/items/models/attempts.ts

```diff
@@ -1,13 +1,42 @@
+import { userBaseSchema, withGroupId } from 'src/app/groups/models/user';
 import { canCurrentUserViewContent, ItemPermWithView } from './item-view-permission';
+import { z } from 'zod';
 
-export const defaultAttemptId = '0'; // to be used as parent attempt for root items
+export const attemptResultSchema = z.object({
+  id: z.string(),
+  allowsSubmissionsUntil: z.coerce.date(),
+  createdAt: z.coerce.date(),
+  endedAt: z.coerce.date().nullable(),
+  latestActivityAt: z.coerce.date(),
+  scoreComputed: z.number(),
+  startedAt: z.coerce.date().nullable(),
+  validated: z.boolean(),
+  userCreator: withGroupId(userBaseSchema).optional(),
+});
 
-interface Result {
+export interface Result {
+  attemptId: string,
   latestActivityAt: Date,
   startedAt: Date|null,
+  score: number,
+  validated: boolean,
+  allowsSubmissionsUntil: Date,
+}
+
+export function resultFromFetchedResult(result: z.infer<typeof attemptResultSchema>): Result {
+  return {
+    attemptId: result.id,
+    latestActivityAt: result.latestActivityAt,
+    startedAt: result.startedAt,
+    score: result.scoreComputed,
+    validated: result.validated,
+    allowsSubmissionsUntil: result.allowsSubmissionsUntil
+  };
 }
```

Comment:
Btw, the idea of mapping level - to keep in interface only needed params? Should we do this practice in other places? Or it's just legacy

Reply:
As notes in the PR description, it is just horrible legacy that we have to get rid of... but in another PR.